msal/src/main/java/com/microsoft/identity/client/GlobalSettings.java

```diff
@@ -0,0 +1,194 @@
+                return;
+            }
+
+            if (mGlobalSettingsSingleton.mPCAMergeAttempted) {
+                listener.onError(new MsalClientException(GLOBAL_INIT_AFTER_PCA_ERROR_CODE,
+                        GLOBAL_INIT_AFTER_PCA_ERROR_MESSAGE));
+                return;
+            }
+
+            setGlobalConfiguration(
+                    initializeGlobalConfiguration(context, configFileResourceId),
+                    listener
+            );
+        }
+    }
+
+    /**
+     * Load the global configuration file using the configuration file and a listener.
+     *
+     * @param configFile Configuration file.
+     * @param listener Handles success and error messages.
+     */
+    @WorkerThread
+    public static void loadGlobalConfigurationFile(@NonNull final File configFile,
+                                                   @NonNull final GlobalSettingsListener listener) {
+        synchronized (mGlobalSettingsSingleton.mGlobalSettingsLock) {
+            if (mGlobalSettingsSingleton.mGlobalSettingsInitialized) {
+                listener.onError(new MsalClientException(GLOBAL_ALREADY_INITIALIZED_ERROR_CODE,
+                        GLOBAL_ALREADY_INITIALIZED_ERROR_MESSAGE));
+                return;
+            }
+
+            if (mGlobalSettingsSingleton.mPCAMergeAttempted) {
+                listener.onError(new MsalClientException(GLOBAL_INIT_AFTER_PCA_ERROR_CODE,
+                        GLOBAL_INIT_AFTER_PCA_ERROR_MESSAGE));
+                return;
+            }
+
+            setGlobalConfiguration(
+                    initializeGlobalConfiguration(configFile),
+                    listener
+            );
+        }
+    }
+
+    @WorkerThread
+    private static void setGlobalConfiguration(@NonNull final GlobalSettingsConfiguration globalConfiguration,
+                                               @NonNull final GlobalSettingsListener listener) {
+        mGlobalSettingsSingleton.mGlobalSettingsConfiguration = globalConfiguration;
+        mGlobalSettingsSingleton.mGlobalSettingsInitialized = true;
+        listener.onSuccess("Global configuration initialized.");
+    }
+
+    @WorkerThread
+    protected PublicClientApplicationConfiguration mergeConfigurationWithGlobal(final @NonNull PublicClientApplicationConfiguration developerConfig) {
+        synchronized (mGlobalSettingsSingleton.mGlobalSettingsLock) {
+            mPCAMergeAttempted = true;
+
+            if (!mGlobalSettingsSingleton.mGlobalSettingsInitialized) {
+                Logger.warn(TAG + "mergeConfigurationWithGlobal",
+                        GlobalSettings.NO_GLOBAL_SETTINGS_WARNING);
+                return developerConfig;
+            }
+
+            developerConfig.mergeGlobalConfiguration(mGlobalSettingsSingleton.mGlobalSettingsConfiguration);
+            developerConfig.validateConfiguration();
+            return developerConfig;
+        }
+    }
+
+    protected static GlobalSettings getInstance() {
```

Comment:
Why protected? I think it can be public.

Reply:
Should this be accessible to developers? I can't think of a reason why they should have access to it.

Reply:
I mean my point is that it probably needs to be either `public` or `package-private` or `private`. The one we pick depends on how we want the API to be accessed. `protected` is usually used if we intend to allow usage of the API by any classes that extend this class. I don't think there are currently any children of `GlobalSettings` so I'm not sure why we would want `protected`.

Reply:
Ah! I see what you mean, I'm removing the access modifier, I really only want this usable in package, and I didn't have plans to make subclasses of GlobalSettings.